openlibrary/plugins/worksearch/code.py

```diff
@@ -1219,8 +1217,9 @@ def work_search(
             facet=facet,
             spellcheck_count=spellcheck_count,
         )
-        response = reply['response'] or ''
-    except (ValueError, OSError) as e:
+        assert reply, "Received None response from run_solr_query"
```

Comment:
I made this a bit more explicit. Getting `None` back should be treated as an exceptional circumstance, and should pass through the exception handling code.